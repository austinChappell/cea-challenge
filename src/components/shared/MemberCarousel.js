import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Card from './Card';

const propTypes = {
  members: PropTypes.arrayOf(PropTypes.object).isRequired,
};

class MemberCarousel extends Component {
  state = {}
  render() {
    const { members } = this.props;
    console.log('members', members);
    return (
      <div className="MemberCarousel">
        <h4>RSVP:</h4>
        <div className="slider">
          {members.map(member => (
            <Card key={member.id}>
              <div>
                <div className="member">
                  <div
                    className="avatar"
                    style={{ backgroundImage: `url(${member.photo.thumb_link})` }}
                  />
                </div>
                <h6>{member.name}</h6>
              </div>
            </Card>
          ))}
        </div>
      </div>
    );
  }
}

MemberCarousel.propTypes = propTypes;

export default MemberCarousel;
