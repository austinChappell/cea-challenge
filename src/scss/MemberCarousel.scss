.MemberCarousel {
  .slider {
    display: flex;
    overflow: auto;
    text-align: center;
    align-items: stretch;
    .member {
      min-width: 100px;
      .avatar {
        margin: 10px auto;
        border-radius: 50%;
        height: 80px;
        width: 80px;
      }
      img {
        border-radius: 50%;
      }
    }
  }
}