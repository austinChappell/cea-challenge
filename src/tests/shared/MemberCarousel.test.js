import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MemberCarousel from '../../components/shared/MemberCarousel';

const props = {
  members: [],
};

describe('The MemberCarousel Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<MemberCarousel {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
