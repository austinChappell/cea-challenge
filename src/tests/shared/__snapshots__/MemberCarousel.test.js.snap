// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MemberCarousel Component should render correctly 1`] = `
<div
  className="MemberCarousel"
>
  <h4>
    RSVP:
  </h4>
  <div
    className="slider"
  />
</div>
`;
